### PR TITLE
Hide OneOfSchema Attributes 

### PR DESCRIPTION
**Pull request by mfleader**
## Changes introduced with this PR

All attributes on the *Schema classes, like `OneOfStringSchema` or `OneOfIntSchema`, in `schema.py` must be public because the schema builder portion of `schema.py` uses metadata analysis of that class's attributes to determine if an instance of the Arcaflow type has been written as specified by the Arcaflow type system (implemented by the *Schema classes and a few other parts of this code base). As a language feature of Python, classes do not have private/hidden attributes. If a private attribute must exist on a *Schema class, then one solution is to encapsulate the required behavior or state on a part of the class that is not used as a part of the metadata analysis. In this pull request I have encapsulated the state we would like to be hidden within method calls on the relevant classes.

Testing for this behavior is not simple, and will have to be introduced in a follow-up pull request.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/arcaflow_plugin_sdk/schema.py [hunk 1]
```diff
@@ -2597,17 +2607,21 @@ class OneOfSchema(_JSONSchemaGenerator, _OpenAPIGenerator):
...
         ),
     ] = "_type"
+    myoneof_type: typing.Annotated[str, _name("One Of Type Schema Name")] = (
+        None
+    )
+
+    def schema_metadata(self) -> OneOfTypeMetadata:
+        raise NotImplementedError(
+            "schema_metadata() is not implemented in the subclass"
+        )
```
[the remark this answers is not in the document]
Comment (webbnh) [reply src/arcaflow_plugin_sdk/schema.py:2624]: If this works as hoped, then we would need `@property` definitions for each of `oneof_type` and `discriminator_type` (and their subclass overriders); with those in place, I believe the rest of the code could be left unchanged (except for the removal of the `__post_init__()` functions).

**Review by webbnh — COMMENTED** [review]
I like Dave's suggestion (assuming it addresses the problem).

File: src/arcaflow_plugin_sdk/schema.py [hunk 2]
```diff
@@ -2597,17 +2607,21 @@ class OneOfSchema(_JSONSchemaGenerator, _OpenAPIGenerator):
...
     discriminator_field_name: typing.Annotated[
         str,
         _name("Discriminator field name"),
         _description(
             "Name of the field used to discriminate between possible values."
         ),
     ] = "_type"
+    myoneof_type: typing.Annotated[str, _name("One Of Type Schema Name")] = (
+        None
+    )
```
[the remark this answers is not in the document]
Comment (webbnh) [reply src/arcaflow_plugin_sdk/schema.py:2619]: I think I'm confused, too.

[the remark this answers is not in the document]
Comment (mfleader) [reply src/arcaflow_plugin_sdk/schema.py:2619]: Oops! I did not mean to leave that in there. I removed it.

File: src/arcaflow_plugin_sdk/schema.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mfleader) [reply src/arcaflow_plugin_sdk/schema.py:2624]: yeah, I considered that implementation, but wasn't 100% sure it would address the issue.

**Review by jaredoconnell — APPROVED** [review]
Looks good. I haven't tested it with a plugin.

File: src/arcaflow_plugin_sdk/schema.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaredoconnell) [reply src/arcaflow_plugin_sdk/schema.py:2619]: Don't forget to push the removal.

**Comment by mfleader** [comment]
For comparison, attached are the manually generated files of the Stress-NG schema from the `main` branch, and this pull request's branch, `oneof-attrs-privatize`. GitHub doesn't want me to upload a `yaml` file, so I rename them to `txt`.
[stressng_schema-main.txt](https://github.com/user-attachments/files/16899166/stressng_schema-main.txt)
[stressng_schema-oneof-attrs-privatize.txt](https://github.com/user-attachments/files/16899169/stressng_schema-oneof-attrs-privatize.txt)

**Review by dbutenhof — APPROVED** [review]
Ah, well; I like the `@property` idea, but I won't object if you don't want to try it.

**Review by webbnh — COMMENTED** [review]
Looks great!

Now, what needs to happen in order to make this change available to https://github.com/arcalot/arcaflow-plugin-stressng/pull/48 (and [#50](https://github.com/arcalot/arcaflow-plugin-stressng/pull/50))?

**Pull request merged**